### PR TITLE
initial setup

### PR DESCRIPTION
**Pull request by iholmdahl**
This contains all scripts and functions that should be necessary to recreate the analysis and figures in the manuscript!

### CODE REVIEW

File: uptakeprojection/R/get_nis_2022.R [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+      )
+  }
+
+  saved <- data %>%
+    arrange(date) %>%
+    filter(!group_category %in% c(
+      "18 – 49 years", "60+ years", "65+ years"
+    )) %>%
+    mutate(
+      group_category = case_when(
```
Comment (swo) [inline uptakeprojection/R/get_nis_2022.R:170]: Should use `case_match()` here

File: uptakeprojection/R/get_nis_2022.R [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+        group_category == "White, non-Hispanic" ~ "White, Non-Hispanic",
+        group_category == "Above poverty, income <$75k" ~ "Above Poverty, Income < $75k",
+        group_category == "Above poverty, income >=$75k" ~ "Above Poverty, Income >= $75k",
+        group_category == "Below poverty" ~ "Below Poverty",
+        group_category == "Unknown income" ~ "Poverty Status Unknown",
+        group_category == "All adults age 18+ years" ~ "Overall",
+        group_category == "18+ years" ~ "Overall",
+        TRUE ~ group_category
+      ),
+      group_name = ifelse(group_name == "Race/Ethnicity (7 level)", "Race/Ethnicity", group_name),
```
Comment (swo) [inline uptakeprojection/R/get_nis_2022.R:192]: this seems like another series of places where `case_match()` would clarify what's going on

**Review by swo — CHANGES_REQUESTED** [review]
- Missing a README, which should have the boilerplate/disclaimers as well as a "Getting Started" section
- This project is not renv enabled, which means that it might be more difficult to write the "getting started" bit. I'll try to get started on my own and see what roadblocks I hit.
- Some small suggested tweaks to the code

**Comment by swo** [comment]
See #2 for my suggestion about how to get the R package into compliance

I tried to `devtools::load_all("uptakeprojection"); source("scripts/setup.R"); source("scripts/fit_slopes.R"); source("scripts/project_uptake.R")`, but I get an error `object 'max_date' not found`, which I think is because of `fit_slopes.R` line 10

**Comment by iholmdahl** [comment]
> I tried to `devtools::load_all("uptakeprojection"); source("scripts/setup.R"); source("scripts/fit_slopes.R"); source("scripts/project_uptake.R")`, but I get an error `object 'max_date' not found`, which I think is because of `fit_slopes.R` line 10

I don't know how the README got deleted but I've added it back in! It clarifies that you only need to run setup.R and then project_uptake.R.

**Comment by swo** [comment]
> I tried to `devtools::load_all("uptakeprojection"); source("scripts/setup.R"); source("scripts/fit_slopes.R"); source("scripts/project_uptake.R")`, but I get an error `object 'max_date' not found`, which I think is because of `fit_slopes.R` line 10

It looks like `max_date` gets defined in `project_uptake.R`, so that would need to be sourced before `fit_slopes.R`. Is there a way around that?

**Comment by iholmdahl** [comment]
> > I tried to `devtools::load_all("uptakeprojection"); source("scripts/setup.R"); source("scripts/fit_slopes.R"); source("scripts/project_uptake.R")`, but I get an error `object 'max_date' not found`, which I think is because of `fit_slopes.R` line 10
> 
> It looks like `max_date` gets defined in `project_uptake.R`, so that would need to be sourced before `fit_slopes.R`. Is there a way around that?

You should not need to run `fit_slopes.R` -- it's sourced by `project_uptake.R`

**Review by swo — APPROVED** [review]

**Pull request merged**